src/components/Publish/Pricing/index.tsx

```diff
@@ -28,6 +29,21 @@ export default function PricingFields(): ReactElement {
     type !== 'free' && setFieldValue('pricing.amountDataToken', 1000)
   }
 
+  // Always update everything when price value changes
+  useEffect(() => {
+    if (type === 'fixed' || type === 'free') return
+
+    const amountOcean =
+      isValidNumber(weightOnOcean) &&
+      isValidNumber(price) &&
+      isValidNumber(weightOnDataToken) &&
+      price > 0
+        ? new Decimal(price).mul(new Decimal(weightOnOcean).mul(10))
+        : new Decimal(initialValues.pricing.amountOcean)
+
+    setFieldValue('pricing.amountOcean', amountOcean)
+  }, [price, weightOnDataToken, weightOnOcean, type, setFieldValue])
+
   // Always update everything when price value changes
```

Comment:
`price` was removed from dependency array so this comment should be adapted. As there are now 2 effects, most likely "update everything" is not true anymore as both effects update different things. Which should be made clear in comments instead of duplicating the comments

Reply:
good point, haven't seen that